gnocchi/rest/aggregates/api.py

```diff
@@ -313,27 +295,48 @@ def groupper(r):
 
             return response
 
+    def get_measures_grouping(self, body, details, fill, granularity,
+                              needed_overlap, references, resources, start,
+                              stop, groupby):
+        def groupper(r):
+            return tuple((attr, r[attr]) for attr in groupby)
+
+        results = []
+        for key, resources in itertools.groupby(resources, groupper):
+            try:
+                results.append({
+                    "group": dict(key),
+                    "measures": self._get_measures_by_name(
+                        resources, references, body["operations"],
+                        start, stop, granularity, needed_overlap, fill,
+                        details=details)
+                })
+            except indexer.NoSuchMetric:
+                pass
+        return results
+
+    def get_measures_grouping_with_history(self, attr_filter, body, details,
+                                           fill, granularity, groupby,
+                                           needed_overlap, references,
+                                           sorts, start, stop):
+        results = []
+        try:
+            grouper = measures_grouper.Grouper(groupby, start, stop, body,
+                                               sorts, attr_filter, references,
+                                               granularity, needed_overlap,
+                                               fill, details)
+            results = grouper.get_grouped_measures()
+        except indexer.NoSuchMetric:
+            pass
+        except indexer.IndexerException as e:
+            api.abort(400, six.text_type(e))
```

Comment:
We dont have a reference to LOG in this scope, I will add one using the 'logging' module, should I use the 'daiquiri' module instead? or is the 'logging' ok?

Also, for this generic exception, catching, in some cases we have already handled exceptions (that are the httpexceptions one) so I will add a new except to ignore these ones to avoid breaking some tests and API compatibilities

Reply:
I think logging is OK since it's used in other places. I don't think you need to introduce anything other than what I suggested since it's just a nicer way of doing it and it would catch anything.

Basicly, just throwing the exception would cause the same behavior anyway, the HTTP 500 error. It's just a matter if we should "leak" the exception message in the response or not. Maybe we shouldn't, what do you think?

Reply:
Thinking about it, we should probably not return the exception, only log it. And you can skip the weboc.HTTPExceptions